google/cloud/internal/future_then_meta.h

```diff
@@ -97,6 +97,47 @@ struct continuation_helper {
   using state_t = future_shared_state<result_t>;
 };
 
+/**
+ * A metafunction to implement `future<T>::%then(Functor&&)`.
+ *
+ * This metafunction implements a number of useful results given a functor type
+ * @p Functor, and the value type @p T of a `future<T>`.
+ *
+ * * First it determines if `Functor` meets the requirements, i.e., that it can
+ *   be invoked with an object of type `future<T>` as its only argument.
+ * * Then in computes the type of the expression `functor(fut)`, where `functor`
+ *   is of type `Functor` and `fut` is of type `future<T>`.
+ * * It determines if the resulting type requires implicit unwrapping because it
+ *   is a `future<U>`.
+ * * It computes the type of future returned from `future<T>::%then(Functor&&)`.
+ *
+ * @tparam Functor the functor to call. It must accept a `future<T>` as its
+ *     single input parameter.
+ * @tparam T the type contained in the input future.
+ */
+template <typename Functor, typename T,
+          typename std::enable_if<is_invocable<Functor, future<T>>::value,
+                                  int>::type = 0>
```

Comment:
Do we need the int here?

Reply:
I guess we could say:

`template<..., typename X = typename std::enable_if<predicate>::type>`

but that seems more verbose than:

`template<..., typename std::enable_if<predicate, int>::type = 0>`

Any reason why you think one is better than the other?  I have no strong opinions (other than I want to type less), but would love the learn about a better way to use `std::enable_if<>`.

Reply:
I was thinking of something like https://godbolt.org/z/HzNegx